MigrationScript/AmsOperationsHelper.ps1

```diff
@@ -0,0 +1,294 @@
+# ########### Header ###########
+# Refer common library file
+. $PSScriptRoot\ConsoleLogger.ps1
+. $PSScriptRoot\UtilityFunctions.ps1
+. $PSScriptRoot\KeyvaultHelperFunctions.ps1
+# #############################
+
+<#
+.SYNOPSIS
+Function to create a new Provider in AMS v2 Monitor.
+
+.PARAMETER subscriptionId
+SubscriptionId of the AMS v2 Monitor.
+
+.PARAMETER resourceGroup
+Resource Group Name of the AMS v2 Monitor.
+
+.PARAMETER monitorName
+AMS v2 Monitor Name.
+
+.PARAMETER request
+request object, it should have Name Property that has provider name and body property which has provider properties. 
+
+.PARAMETER logger
+logger object. 
+
+.EXAMPLE
+$requestObj = @{
+				Name = $providerName
+				body = @{
+					properties = @{
+						providerSettings = @{
+							providerType = $providerType
+							hostname = $Hostname
+							dbName = $DbName
+							sqlPort = $SqlPort
+							dbUsername = $DbUsername
+							dbPassword = $DbPassword
+						}
+					}
+				}
+			}
+$logger = New-Object ConsoleLogger
+PutAmsV2Provider -subscriptionId $subscriptionId -resourceGroup $resourceGroupName -monitorName $monitorName -request $requestObj -logger $logger;
+#>
+function PutAmsV2Provider([string]$subscriptionId, [string]$resourceGroup, [string]$monitorName, $request, $logger) {
+	$rawToken = Get-AzAccessToken -ResourceTypeName Arm;
+    $armToken = $rawToken.Token;
+    $apiVersion = "2021-12-01-preview";
+
+	$headers = @{
+        "Content-Type" = "application/json"
+        "Authorization" = "Bearer $armToken"
+    };
+
+	$providerName = $request.name;
+    $bodyStr = $request.body | ConvertTo-Json -Depth 3;
```

Comment:
Lets correct the formating, use anything you like, but for this one indent seems to be off.

Reply:
The formatting seems to be okay in my local setup

Reply:
Something messing it up then in the web view, okay lets ignore it for now.

Reply:
Okay so resolving comment for now. We'll keep a watch